tcms/testcases/migrations/0010_remove_bug.py

```diff
@@ -1,20 +1,54 @@
+import glob
+import json
+
 from django.db import migrations
 
 
 def forward_copy_data(apps, schema_editor):
     bug_model = apps.get_model('testcases', 'Bug')
     link_reference_model = apps.get_model('linkreference', 'LinkReference')
 
-    for bug in bug_model.objects.all():
-        if not bug.case_run_id:
-            continue
+    link_reference_ids_file_name = '/tmp/kiwitcms-testcases-migrations-0010-\
+new-LinkReference-IDs'  # nosec:B108:hardcoded_tmp_directory
+    with open(link_reference_ids_file_name, 'w') as link_reference_ids_file:
```

Comment:
Writing the IDs is independent of collecting them & writing the bug records, you don't need to have this file open right until the end of this block. Just open the file & dump the ids once there.